utils/build-script

```diff
@@ -497,6 +504,10 @@ the number of parallel build jobs to use""",
         dest="build_jobs",
         default=multiprocessing.cpu_count())
 
+    parser.add_argument("--darwin-xcrun-toolchain",
+                        help="the name of the toolchain to use on Darwin",
+                        default="default")
+
```

Comment:
Sorry, but this is not a backwards-compatible change.  Previously, `--darwin-xcrun-toolchain ` was an argument that was passed after `--`:

```
$ ./utils/build-script -RT -- --darwin-xcrun-toolchain=foobar --other --flags
```

In order for the argument parsing to catch it now, it needs to be passed before `--`:

```
$ ./utils/build-script -RT --darwin-xcrun-toolchain=foobar --  --other --flags
```

I'm totally on board with breaking backward compatibility *once* when the refactoring is done, but not continuously throughout the process.

Thus, for staging, it seems like what we need to do is to preprocess the command line like this:

1. Filter out `--darwin-xcrun-toolchain.*` arguments from the command line into a separate list.

2. Append that list right before `--`.

3. Pass the new list to `parser.parse_args()`.

I think this pattern will appear over and over again as you migrate handling of other command-line arguments to Python, so it would make sense to make it a reusable function.